### PR TITLE
fix for two flaky UI Tests 

### PR DESCRIPTION
**Pull request by pascal-brause**
## Description
This should stable the two UI Tests:
- ENAUITests_09_TraceLocations.test_events_in_contact_journal
- ENAUITests_10_CheckIns.testCheckinInfoScreen_navigate_to_dataPrivacy

For the first (09), I refactored the way how we check f there are two locations in the overview cell by adding an index-based accessibility identifier.
The second (10) should work now because i scroll down 10 times to ensure, we get to the bottom of the table view to hit the privacy cell.

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-6896

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/View Helpers/AccessibilityIdentifiers.swift [hunk 1]
```diff
@@ -102,6 +102,8 @@ enum AccessibilityIdentifiers {
 			static let checkinRiskLevelLow = "AppStrings.ContactDiary.Overview.CheckinEncounter.titleLowRisk"
 			static let checkinRiskLevelHigh = "AppStrings.ContactDiary.Overview.CheckinEncounter.titleHighRisk"
 			static let checkinTableView = "AppStrings.ContactDiary.Overview.CheckinEncounter.tableView"
+			static let person = "ContactDiary_Overview_personEntry-%i"
+			static let location = "ContactDiary_Overview_locationEntry-%i"
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/View Helpers/AccessibilityIdentifiers.swift:106]: Please use `%d` here:
```suggestion
			static let person = "ContactDiary_Overview_personEntry-%d"
			static let location = "ContactDiary_Overview_locationEntry-%d"
```

File: src/xcode/ENA/ENAUITests/CheckIns/ENAUITests_10_CheckIns.swift [hunk 2]
```diff
@@ -80,12 +80,23 @@ class ENAUITests_10_CheckIns: XCTestCase {
...
+			return
+		}
+		
+		let maxTries = 10
+		var actualTry = 0
+		while lastCell.isHittable == false && actualTry < maxTries {
+			app.swipeUp()
+			actualTry += 1
+		}
+		lastCell.tap()
```
Comment (whiskey) [inline src/xcode/ENA/ENAUITests/CheckIns/ENAUITests_10_CheckIns.swift:97]: Do we really have to swipe through? If we do have the `AccessibilityIdentifier` we can omit the swiping:
```
let cell = app.cells[$lastCellsID]
XCTAssertTrue(waitForExistence(timeout: .short)) // not sure if needed
cell.tap()
```

File: src/xcode/ENA/ENAUITests/TraceLocations/ENAUITests_09_TraceLocations.swift [hunk 3]
```diff
@@ -372,20 +372,15 @@ class ENAUITests_09_TraceLocations: XCTestCase {
...
-			eventcount[0] += texts["\(event0), \(traceLocations_checked_in[event0] ?? "")"].exists ? 1 : 0
-			eventcount[1] += texts["\(event1), \(traceLocations_checked_in[event1] ?? "")"].exists ? 1 : 0
-			eventcount[2] += texts["\(event2), \(traceLocations_not_checked_in[event2] ?? "")"].exists ? 1 : 0
-		}
 		
-		XCTAssertTrue(eventcount[0] == 1)
-		XCTAssertTrue(eventcount[1] == 1)
-		XCTAssertTrue(eventcount[2] == 0)
+		// Get the first overview table view cell.
+		let overviewCell = app.tables.firstMatch.cells.element(boundBy: 1)
```
Comment (whiskey) [inline src/xcode/ENA/ENAUITests/TraceLocations/ENAUITests_09_TraceLocations.swift:379]: Would be nice to use more AccessibilityIdentifiers here (and in general). We should address this to the team.

File: src/xcode/ENA/ENAUITests/TraceLocations/ENAUITests_09_TraceLocations.swift [hunk 4]
```diff
@@ -372,20 +372,15 @@ class ENAUITests_09_TraceLocations: XCTestCase {
...
+		XCTAssertTrue(app.navigationBars[AccessibilityLabels.localized(AppStrings.ContactDiary.Overview.title)].waitForExistence(timeout: .medium))
 
-		// count the number of entries on the screen
-		var eventcount = [0, 0, 0]
-		for i in 0...(app.cells.count - 1) {
-			let texts = app.cells.element(boundBy: i).staticTexts
-			eventcount[0] += texts["\(event0), \(traceLocations_checked_in[event0] ?? "")"].exists ? 1 : 0
-			eventcount[1] += texts["\(event1), \(traceLocations_checked_in[event1] ?? "")"].exists ? 1 : 0
-			eventcount[2] += texts["\(event2), \(traceLocations_not_checked_in[event2] ?? "")"].exists ? 1 : 0
-		}
```
Comment (whiskey) [inline src/xcode/ENA/ENAUITests/TraceLocations/ENAUITests_09_TraceLocations.swift:384]: Good example for a unit but not a UI test.

File: src/xcode/ENA/ENAUITests/CheckIns/ENAUITests_10_CheckIns.swift (hunk 2 above)
Comment (pascal-brause) [reply src/xcode/ENA/ENAUITests/CheckIns/ENAUITests_10_CheckIns.swift:97]: Normally I would agree, but before we used it and it did not work properly.

File: src/xcode/ENA/ENAUITests/TraceLocations/ENAUITests_09_TraceLocations.swift (hunk 3 above)
Comment (pascal-brause) [reply src/xcode/ENA/ENAUITests/TraceLocations/ENAUITests_09_TraceLocations.swift:379]: So set an identifier like I did for the places in that PR?

File: src/xcode/ENA/ENAUITests/CheckIns/ENAUITests_10_CheckIns.swift (hunk 2 above)
Comment (naveeddotio) [reply src/xcode/ENA/ENAUITests/CheckIns/ENAUITests_10_CheckIns.swift:97]: I agree, wait for existence not always works, somehow, the whole eco-system around UI testing is unfortunately a mess.

Comment (whiskey) [reply src/xcode/ENA/ENAUITests/CheckIns/ENAUITests_10_CheckIns.swift:97]: I think it's a bug on our side but if this works atm, let's keep it.

File: src/xcode/ENA/ENAUITests/TraceLocations/ENAUITests_09_TraceLocations.swift (hunk 3 above)
Comment (whiskey) [reply src/xcode/ENA/ENAUITests/TraceLocations/ENAUITests_09_TraceLocations.swift:379]: If feasible, then yes 👍

Comment (pascal-brause) [reply src/xcode/ENA/ENAUITests/TraceLocations/ENAUITests_09_TraceLocations.swift:379]: i will ad it, no problem

**Review by whiskey — APPROVED** [review]

**Review by naveeddotio — APPROVED** [review]

**Pull request merged**